### DISCUSSION TITLE
language setting

### DISCUSSION THREAD

**Discussion by johanpeters** [akeeba/panopticon#323] [Q&A (Community Support)] [answered]
Hello,
when i change the default language in settings. the first time i changed it the language of the site is switched.
now i would like to switch it back to english. on save the saved message is in the new language but the rest of the site is keeping the previous language.
clear browser cache, same result.

**Comment by nikosdion** [comment]
The configuration option only changes the **default** language. It does not change the application language. This is determined automatically using your browser's preferences. If your browser's preferred language is Dutch then you will be seeing the interface in Dutch.

This is a deliberate feature. You are accessing Panopticon through a login. As such, using a shared computer is out of the question for security reasons. Therefore, you are in complete control of your browser and its settings, including the language preference. Which means that instead of asking you to provide your preferred language we should be using your browser's language preference.

Asking users for a preferred language only makes sense when you have guest sessions, in which case it is plausible that they are using a shared computer whose language settings may not be under their control. In other words, having a language switcher makes sense only for the public part of a website. It does not really make sense when you are behind a login.

**Reply by nikosdion** [reply, marked as answer]
Right now, the language is determined by the settings of the user's browser. I can see your use cases where this does not work very well for you, and the fact that this also means that the emails are always sent in the default language.

I have opened issue #326 which will address this issue in version 1.0.5.

Each user will be able to set their own language preference in their user account. This means that _after they are logged in_ we will use whichever language is specified there. The default will be a “Use default” kind of setting which works as previously: it uses the browser's preferred language.

For emails, the preferred language will be the one set up for each user. If the user has no preferred language, the default language configured in the system configuration page will be used. The caveat is that all CC'ed email addresses will also be emailed in that default language since they do not correspond to users _even if_ you are using an email address which corresponds to a different user account.

**Comment by Martin511** [comment]
I do not really agree, Nicholas.
If I want to translate Panopticon for example, it would be good to change the displayed language of my choice, independant from my browser´s settings.
I absolutely appreciate to have control about language settings in applications independant from my browser´s setting.
Might be my very own opinion.

**Comment by johanpeters** [comment]
thanks for clearing this up.. it would be nice do have a option to set the language per user.
based on the fact that you can delicate accounts to for example customers i can think about situations that it would be nice to have the option to set the language fixed on user account level.
default browser settings (current way of working) but with option to force the language general of user specific.

**Comment by brianteeman** [comment]
So the emails will only ever be sent in the default language and not in the specific language of a user as panopticon has no way of knowing that the language is for each user. ??

**Reply by nikosdion** [reply]
Correct. AWF was designed as a single natural language framework. I have to think about whether this can reasonably change. My primary concern is that I don't want to introduce any change which might screw up Akeeba Backup for WordPress. I've gone through Hell and back with WordPress 6.3 over the last couple of months. 0/10, would not recommend.

**Reply by brianteeman** [reply]
Thanks for clarifying that
